storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2037,10 +2036,6 @@ static bool innodb_init_param(void) {
     }
   }
 
-  os_innodb_umask = (ulint)0664;
-
-  os_file_set_umask(my_umask);
```

Comment:
where is this set before? Also, can you please check if this os_file-set_umask() is removed from innodb_init_param in InnoDB sources?

Reply:
No, I mean the code in xtrabackup.cc
innob_init_params is duplicated in xtrabackup.cc and the os_file_set_umask() is removed this function in xtrbackup.cc 

By removing this mask, the on-disk file mode is changed from 0664 to 0614.

But if you bring back the set_umask(), prepare will fail with assertion because we reboot (and so reinit) innodb multiple times.

We have to allow the double incase of pxb. May be introduce a de-init function of umask() and it allows the reinit.

Reply:
added the umask resetting option for xtrabackup in the latest commit, please take a look